cpp/open3d/t/geometry/TriangleMesh.cpp

```diff
@@ -1252,6 +1252,143 @@ TriangleMesh TriangleMesh::SelectByIndex(const core::Tensor &indices) const {
     return result;
 }
 
+namespace {
+
+/// This is a templatized local version of RemoveDuplicates.
+/// Use of templates allows us to use common code for different
+/// datatype possibilities for vertex.positions, namely, float32, float64,
+/// and triangle.indices, namely, int32, int64.
+/// This function first computes two maps
+/// (MapA.) From vertex coordinates to their indices.
+//          This is implemented using unordered_map.
+/// (MapB.) From the vertex indices before duplicate removal to indices after
+/// removal.
+///          This is implemented using std::vector.
+///
+/// MapA allows the function to detect duplicates and MapB allows it update
+/// triangle indices. During the computation of the above maps unique vertex
+/// attributes including positions are moved to beginning of the corresponding
+/// attribute tensor.
+///
+/// Caveats:
+///   The unordered_map approach is fast in general but if the inputs are close,
+///   the hashing could cause problems and it might be good to consider a
+///   sorting based algorithms, which can do a better job of comparing the
+///   double values.
+/// \param mesh The mesh from which duplicate vertices are to be removed.
+template <typename Coord_t, typename Tindex_t>
+TriangleMesh &RemoveDuplicateVerticesWorker(TriangleMesh &mesh) {
+    // Algorithm based on Eigen implementation
+    if (!mesh.HasVertexPositions()) {
+        utility::LogWarning(
+                "TriangeMesh::RemoveDuplicateVertices: No vertices present, "
+                "ignoring.");
+        return mesh;
+    }
+
+    typedef std::tuple<Coord_t, Coord_t, Coord_t> Point3d;
+    std::unordered_map<Point3d, size_t, utility::hash_tuple<Point3d>>
+            point_to_old_index;
+
+    auto vertices = mesh.GetVertexPositions().To(mesh.GetDevice()).Contiguous();
+    auto orig_num_vertices = vertices.GetLength();
+    const auto vmask_type = core::Bool;
+    using vmask_itype = bool;
+    core::Tensor vertex_mask =
+            core::Tensor::Zeros({orig_num_vertices}, vmask_type);
+    using Length_t = decltype(orig_num_vertices);
+    std::vector<Length_t> index_old_to_new(orig_num_vertices);
+
+    // REM_DUP_VERT_STEP 1:
+    // Compute map from points to old indices, and use a counter
+    // to compute a map from old indices to new unique indices.
+    const Coord_t *vertices_ptr = vertices.GetDataPtr<Coord_t>();
+    vmask_itype *vertex_mask_ptr = vertex_mask.GetDataPtr<vmask_itype>();
+    Length_t k = 0;                                     // new index
+    for (Length_t i = 0; i < orig_num_vertices; ++i) {  // old index
+        Point3d coord =
+                std::make_tuple(vertices_ptr[i * 3], vertices_ptr[i * 3 + 1],
+                                vertices_ptr[i * 3 + 2]);
+        if (point_to_old_index.find(coord) == point_to_old_index.end()) {
+            point_to_old_index[coord] = i;
+            index_old_to_new[i] = k;
+            ++k;
+            vertex_mask_ptr[i] = 1;
```

Comment:
nit: 1 -> true, since it is a bool value here. (it is 1 in other functions above as we use int there for the prefix_sum approach).